### PR TITLE
UMICollapse module: Drop external dependencies from UMICollapse module tests

### PR DESCRIPTION
**Pull request by MatthiasZepper**
Currently, the [UMICollapse module](https://nf-co.re/modules/umicollapse) tests run on data that is generated in the test `setup{}` with external dependencies (`umitools/extract`, `bwa/index`, `bwa/mem`, `samtools/index`). While this is no problem here in the modules' repository, those tests will fail on the pipeline level, if the additional modules are not installed.

While it may, on the long-term, be useful to devise a CI step detecting and installing missing modules for tests ([Leon Rauschning made a neat proposal for this](https://nfcore.slack.com/archives/C049MBCEW06/p1731664424635429?thread_ts=1731521149.600189&cid=C049MBCEW06)), I have now, due to time contraints, opted to add the required files to the test datasets ([test-datasets/1390](https://github.com/nf-core/test-datasets/pull/1390)) and change the tests accordingly (**this PR**).

I have also updated the `umi-tools dedup` tests, since they so far used inputs without UMIs.  

## PR checklist

- [x] This comment contains a description of changes (with reason).
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/umicollapse/tests/main.nf.test [hunk 1]
```diff
@@ -13,57 +13,16 @@ nextflow_process {
     tag "bwa/mem"
```
Comment (SPPearce) [inline modules/nf-core/umicollapse/tests/main.nf.test:13]: We can remove the tags then - can't comment on the lines above.

**Review by SPPearce — APPROVED** [review]
Looks much better.
Think you just need to swap to using the nft-bam plugin to test the bam files then should be good.

File: modules/nf-core/umicollapse/tests/main.nf.test (hunk 1 above)
Comment (MatthiasZepper) [reply modules/nf-core/umicollapse/tests/main.nf.test]: Done. Thanks a lot for the eagle eye!

Comment (SPPearce) [reply modules/nf-core/umicollapse/tests/main.nf.test]: That'd be the point of reviewing ;)

**Pull request merged**